### PR TITLE
generate generic urls for sites

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by miltontony** [comment]
@smn @Rizziepit review please?
I've started refactoring stuff. My plan is to move the MC codebase to a place where adding a new project type (like Molo/Django) doesn't in away touch the i.org stuff and is much cleaner

**Comment by Rizziepit** [comment]
+1

File: unicoremc/models.py [hunk 1]
```diff
@@ -220,9 +228,25 @@ def all_repos(self):
     def get_state_display(self):
         return states.ProjectWorkflow(instance=self).get_state()
 
+    def get_generic_domain(self):
+        hub = 'qa-hub' if settings.DEPLOY_ENVIRONMENT == 'qa' else 'hub'
+        return '%(app_id)s.%(hub)s.unicore.io' % {
+            'app_id': self.app_id,
+            'hub': hub
+        }
```
Comment (smn) [inline unicoremc/models.py:236]: Eventually we want these to be `qa-hub` & `hub` aliases to the actual domains being things like:

``` python
import random


def do_me_a_domain():
    adjectives = [
        "autumn", "hidden", "bitter", "misty", "silent", "empty", "dry",
        "dark", "summer", "icy", "delicate", "quiet", "white", "cool",
        "spring", "winter", "patient", "twilight", "dawn", "crimson", "wispy",
        "weathered", "blue", "billowing", "broken", "cold", "damp", "falling",
        "frosty", "green", "long", "late", "lingering", "bold", "little",
        "morning", "muddy", "old", "red", "rough", "still", "small",
        "sparkling", "throbbing", "shy", "wandering", "withered", "wild",
        "black", "young", "holy", "solitary", "fragrant", "aged", "snowy",
        "proud", "floral", "restless", "divine", "polished", "ancient",
        "purple", "lively", "nameless"
    ]
    nouns = [
        "waterfall", "river", "breeze", "moon", "rain", "wind", "sea",
        "snow", "lake", "sunset", "pine", "shadow", "leaf", "dawn", "glitter",
        "forest", "hill", "cloud", "meadow", "sun", "glade", "bird", "brook",
        "butterfly", "bush", "dew", "dust", "field", "fire", "flower",
        "feather", "grass", "haze", "mountain", "night", "pond", "darkness",
        "snowflake", "silence", "sound", "sky", "shape", "surf", "thunder",
        "violet", "water", "wildflower", "wave", "water", "resonance", "sun",
        "wood", "dream", "cherry", "tree", "fog", "frost", "voice", "paper",
        "frog", "smoke", "star", "morning", "firefly",
    ]
    return "%s-%s-%s" % (
        random.choice(adjectives),
        random.choice(nouns),
        random.randint(1, 1000))

print do_me_a_domain()
```

```
ancient-voice-574
frosty-moon-129
broken-rain-212
divine-mountain-562
hidden-thunder-557
snowy-resonance-375
crimson-shadow-479
proud-field-560
weathered-smoke-952
small-wind-466
```

Comment (Rizziepit) [inline unicoremc/models.py:236]: All the +1 for this

Comment (smn) [inline unicoremc/models.py:236]: Dibs on `small-wind` and `hidden-thunder` :grinning:

**Pull request merged**